Source/Welding.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using UnityEngine;
+using System.Reflection;
+using PreFlightTests;
+using TestScripts;
+
+namespace Kaboom
+{
+    public class Welding
+    {
+        Vessel vessel;
+        Part part;
+        public Welding(Vessel vessel, Part part)
+        {
+            this.vessel = vessel;
+            this.part = part;
+        }
+
+        public void MergeParts(bool compress)
+        {
+            if (vessel.rootPart == part)
+            {
+                ScreenMessages.PostScreenMessage("You cannot weld the root part!");
+                return;
+            }
+
+            var wData = LoadWeldingData();
+            if (wData == null)
+                return;
+
+            PerformWeld(wData, compress);
+        }
+
+        private WeldingData LoadWeldingData(bool silent = false)
+        {
+            /**********************
+             * 
+             *  (root)-...-LPA==KGP==LPB
+             * 
+             *     LPA==LPB
+             * 
+             **********************/
+
+            var wData = new WeldingData();
+            wData.KaboomGluedPart = part;
+
+
+            if (part.attachNodes.Count == 2)
+            {
+                
+                wData.LinkedPartA = part.parent;
+
+                foreach(var n in part.attachNodes)
+                    if (n.attachedPart != part.parent)
+                        wData.LinkedPartB = n.attachedPart;
+            }
+
+            if (wData.LinkedPartA == null || wData.LinkedPartB == null)
+            {
+                if (!silent)
+                    ScreenMessages.PostScreenMessage("This part need to have 2 parts on attachment nodes");
+                return null;
+            }
+
+            if (wData.KaboomGluedPart == vessel.rootPart)
+            {
+                if (!silent)
+                    ScreenMessages.PostScreenMessage("This part is the root part!  Cancelling");
+                return null;
+            }
+
+            return wData;
+        }
+
+
+        private Vector3 GetOffset(WeldingData wData)
```

Comment:
Welding.GetOffset based on
https://github.com/UmbraSpaceIndustries/Konstruction/blob/18b39deb9eb0ffe6c2c66da97629d0f631c9d1da/Source/Konstruction/Konstruction/Welding/ModuleWeldablePort.cs#L271
doesn't work properly

Reply:
have to start somewhere though. 👍